locations/spiders/lidl_de.py

```diff
@@ -1,84 +1,63 @@
-                    open_time=match.group(1),
-                    close_time=match.group(2),
-                    time_format="%H:%M",
-                )
-
-        return opening_hours.as_opening_hours()
-
+    start_urls = ["https://www.lidl.de/f/"]
     def parse_details(self, response):
-        properties = {}
-        if response.status == 200:
-            json_data = re.search(r"salePoints = eval\((.*?)\);", response.text)
-            if json_data:
-                json_data = json.loads(json_data.groups(1)[0])
-                for store in json_data:
-                    if store["country"] == "DE":
-                        properties = {
-                            "country": store["country"],
-                            "ref": store["ID"],
-                            "name": store["name"],
-                            "street": store["STREET"],
-                            "postcode": store["ZIPCODE"],
-                            "city": store["CITY"],
-                            "lat": store["X"],
-                            "lon": store["Y"],
-                        }
-                        hours = self.parse_hours(store["parsedopeningTimes"]["regular"])
-                        if hours:
-                            properties["opening_hours"] = hours
 
-                        yield GeojsonPointItem(**properties)
+        lidlShops = response.css('.ret-o-store-detail')
+
+        for shop in lidlShops:
+            shopAddress = shop.css('.ret-o-store-detail__address::text').extract()
+            street = shopAddress[0]
+            postalCode = shopAddress[1].split()[0]
+            city = shopAddress[1].split()[1]
+
+            openingHours = shop.css('.ret-o-store-detail__opening-hours::text').extract()
+            shopOpeningHours = {}
+            for openingHour in openingHours:
+                if(openingHour.split()):
+                    day = openingHour.split()[0]
+                    hours = openingHour.split()[1]
+                    shopOpeningHours[day] = hours
+
+            services = response.css('.ret-o-store-detail__store-icon-wrapper')[0]
+            link = services.css('a::attr("href")').get()
+            coordinates = link.split('pos.')[1].split('_L')[0]
+            latitude = coordinates.split('_')[0]
+            longitude = coordinates.split('_')[1]
+
+            properties = {
+                "ref": latitude+longitude,
+                "country": "Germany",
+                "name": "Lidl",
+                "street": street,
+                "postcode": postalCode,
+                "city": city,
+                "opening_hours": shopOpeningHours,
+                "lat": latitude,
+                "lon": longitude
+            }
+            yield GeojsonPointItem(**properties)
 
     def parse(self, response):
         # Read all provinces
-        province = response.xpath('//dl[@class="store-listing"]')
+        provinces = response.css('.ret-o-store-detail-content')
```

Comment:
Did you mean `ret-o-store-detail-content` or `ret-o-store-detail-cities`? I don't think it matters anyway, we can just go straight to the `ret-o-store-detail-city` from the response.

I'd also have used a [CrawlSpider](https://docs.scrapy.org/en/latest/topics/spiders.html?highlight=CrawlSpider#crawlspider-example), but that's not important now the code is written.